plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/N4JSEnumValidator.java

```diff
@@ -106,6 +116,49 @@ public void checkEnumLiterals(N4EnumDeclaration n4EnumDeclaration) {
 				}).collect(Collectors.toSet());
 	}
 
+	/**
+	 * See N4JS Specification, Req. IDE-41, Nr. 6.
+	 */
+	@Check
+	public void checkUsageOfStringBasedEnum(IdentifierRef identRef) {
+		final IdentifiableElement id = identRef.getId();
+		if (id == null || id.eIsProxy()) {
+			return;
+		}
+		if (!(id instanceof TEnum)) {
+			return;
+		}
+		final TEnum tEnum = (TEnum) id;
+		if (!AnnotationDefinition.STRING_BASED.hasAnnotation(tEnum)) {
+			return;
+		}
+		// we now have an IdentifierRef pointing to a string-based enum ...
+		final EObject parent = identRef.eContainer();
+		final ParameterizedPropertyAccessExpression parentPAE = parent instanceof ParameterizedPropertyAccessExpression
+				? (ParameterizedPropertyAccessExpression) parent : null;
+		final IdentifiableElement prop = parentPAE != null ? parentPAE.getProperty() : null;
+		if (prop != null) {
+			if (prop.eIsProxy()) {
+				// there will be an error for the unresolved property access, so any error shown below would be an
+				// unnecessary duplicate error
+				return;
+			}
+			if (prop instanceof TEnumLiteral && prop.eContainer() == tEnum) {
```

Comment:
Is this check really necessary? `prop.eContainer() == tEnum)`

Reply:
It should(!) not be required, but I added it to be on the safe side and for readability.
But I now changed the entire condition to a different, still more readable condition.